Userland/Libraries/LibWeb/Bindings/WindowObject.cpp

```diff
@@ -650,6 +651,17 @@ JS_DEFINE_NATIVE_FUNCTION(WindowObject::screen_y_getter)
     return JS::Value(impl->screen_y());
 }
 
+JS_DEFINE_NATIVE_FUNCTION(WindowObject::origin_getter)
+{
+    auto* impl = TRY(impl_from(vm, global_object));
+    auto origin = impl->associated_document().origin();
+    String result = "null";
+    if (!origin.host().is_empty()) {
+        result = String::formatted("{}://{}", origin.protocol(), origin.host());
+    }
```

Comment:
This needs to be abstracted into `Origin::serialize()`, following the spec: https://html.spec.whatwg.org/multipage/origin.html#ascii-serialisation-of-an-origin